app/controllers/report_controller/saved_reports.rb

```diff
@@ -156,10 +156,10 @@ def set_saved_reports_condition(rep_id = nil)
 
     # Admin users can see all saved reports
     unless admin_user?
-      cond[0] << " AND miq_group_id=?"
-      cond.push(current_group_id)
+      cond[0] << " AND miq_group_id IN (?)"
+      cond.push(current_user.miq_groups.collect(&:id))
```

Comment:
Maybe change this to do concat and then you won't need to flatten the array before returning it.